torch/autograd/grad_mode.py

```diff
@@ -395,3 +395,38 @@ def __enter__(self) -> None:
 
     def __exit__(self, *args) -> None:
         torch._C._autograd._unsafe_set_version_counter(self.tensor, self.prev_version)
+
+
+class _unsafe_preserve_version_counters(_DecoratorContextManager):
```

Comment:
Can we just do that within the function above with `Union[Tensor, tuple(Tensor, ...)]` ?

Reply:
good idea. will extend the existing function